### PR TITLE
fix(stacked-horde): validate cumulant source bounds

### LINKED ISSUE #579: StackedLinearHorde silently clips out-of-range cumulant indices
## Summary

`StackedLinearHorde.update` gathers cumulants with a JAX array index before checking that the caller supplied every configured channel. JAX clips an out-of-range positive gather index instead of raising, so a demon can silently train on the last available channel.

Concrete behavior on current `main` (`06b0fbd75467081d6f034c828d1dd8bcc18eea1a`):

```text
cumulant_indices=(0, 2)
source=[1, 99, 4] -> td_errors=[1, 4]   # expected channels
source=[1, 99]    -> td_errors=[1, 99]  # index 2 silently clips to channel 1
```

With `step_size=0.5` and features `[1, 0, 0]`, the second demon's first weight becomes `49.5` instead of rejecting the missing channel. The same config is publicly reachable through `nexting_spec(...)`, `StackedLinearHorde`, and `run_stacked_horde_scan`.

Negative and boolean `cumulant_indices` are also accepted today; negative indices wrap and booleans become integer channels.

A tests-first run against the current implementation produced **6 failures and 61 passes** before the repair.

## Expected behavior

- Require every configured cumulant index to be a nonnegative built-in integer.
- Require one-step `cumulant_source` inputs to be rank-one.
- Require scan `cumulant_sources` inputs to be rank-two.
- Reject a source whose final axis does not contain the maximum configured index before any gather or state update.
- Preserve valid direct, JIT-compiled, and scan updates unchanged.

## Scope

- `alberta_framework/core/stacked_horde.py`
- `tests/test_stacked_horde.py`

This is a bounded input-domain repair. It does not change valid TD(lambda) equations, state layout, public serialization keys, benchmark thresholds, frozen protocols, seeds, scientific claims, `CHANGELOG.md`, or any `outputs/**` artifact.

## Ownership check

A complete current scan covered **57 open ASI PRs / 174 changed-file rows** and found no overlap with either target path. A separate scan covered **64 open ASI issues**, including claim comments on matching issues, and found no issue matching Stacked Horde cumulant-source bounds.

## Contribution provenance

- AI assistance: yes
- AI provider/model: xAI / grok-4.6; OpenAI / gpt-5.6-sol
- Client / agent tooling: Grok Build; Claude Code via CLIProxyAPI
- Attribution status: self-reported

### PR DESCRIPTION
**Pull request by ss251**
Fixes #579.

## What changed

`StackedLinearHorde` now validates the complete cumulant-source shape before JAX gathers configured channels:

- every `cumulant_indices` entry must be a nonnegative exact built-in integer;
- one-step `cumulant_source` inputs must be rank one;
- scan `cumulant_sources` inputs must be rank two;
- the source's final axis must contain the maximum configured cumulant index before any gather or state update;
- the same static-shape checks apply to direct, `jax.jit(...)`, and `jax.lax.scan` paths.

This prevents JAX's positive out-of-range gather clipping from silently training a demon on the final available channel. Negative indices no longer wrap, and booleans / NumPy integer aliases no longer become configuration channels.

Valid TD(lambda) equations, state layout, update masking, serialization keys, and nexting behavior are unchanged.

## Reproduction

On the pre-change implementation:

```text
cumulant_indices=(0, 2)
source=[1, 99, 4] -> td_errors=[1, 4]
source=[1, 99]    -> td_errors=[1, 99]  # index 2 silently clips to channel 1
```

The failing-first focused run produced **6 failures and 61 passes**. The failures demonstrated acceptance of negative/boolean/NumPy indices, silent one-step and scan clipping, and a later broadcast failure for a rank-two one-step source instead of a boundary error.

## Verification

Final base: `80f1ace55108d38b9a25e6339e1efecf3f45c95c`

Exact head: `c82b915fc0577f5880f0ba66ad8c7a3ba72cd071`

- focused module: **68 passed**;
- direct and JIT short-source rejection: passed;
- valid wide-source channel-selection proof: passed;
- one-step rank validation: passed;
- scan short-source validation: passed;
- Ruff on changed source and tests: passed;
- strict mypy on changed production source: passed;
- `git diff --check origin/main...HEAD`: clean;
- exact-head factory proof: recorded green for `c82b915fc`.

## Ownership and scientific integrity

A complete current scan covered **61 open ASI PRs / 182 changed-file rows** with zero overlap on either target path. A separate scan covered **57 open issues** and matching claim comments; only #579 claims Stacked Horde cumulant-source bounds.

No `outputs/**` file, frozen protocol, benchmark threshold, seed, scientific claim, or registered five-claim evidence source changed. `alberta-evidence-status` remains inherited-invalid from unrelated source-hash drift already present on `main`; this PR does not suppress, regenerate, revalidate, or promote any evidence.

## Contribution provenance

- AI assistance: yes
- AI provider/model: xAI / grok-4.6; OpenAI / gpt-5.6-sol
- Client / agent tooling: Grok Build; Claude Code via CLIProxyAPI
- Attribution status: self-reported

— [sol-stacked-horde-cumulant-bounds]

### CODE REVIEW

**Review by lalalune — APPROVED** [review]
Maintainer review: the claimed defect is real, the current head fixes it without broadening the contract unexpectedly, and the focused regression coverage matches the changed boundary. Approved subject to required checks and integration-order revalidation.

**Comment by ss251** [comment]
Maintainer PR #608 has now merged the same #579 configuration boundary in the same target files. This PR is therefore superseded; requesting closure. I will not push another head to this branch.

AI provider/model: OpenAI / gpt-5.6-sol
Client / agent tooling: Claude Code via CLIProxyAPI
Attribution status: self-reported

**Comment by lalalune** [comment]
Closing as superseded by merged #608, which implements the same cumulant-source validation with broader constructor, serialization, rank, and execution-path coverage. This branch is now conflicting, and the author also requested closure.
